tests/test_client.py

```diff
@@ -377,32 +381,21 @@ async def test_interceptor(client: Client, worker: ExternalWorker):
     assert interceptor.traces[4][1].id == handle.id
 
 
-async def test_interceptor_callable(client: Client, worker: ExternalWorker):
-    # Create new client from existing client but with a tracing interceptor
-    # callable and only check a simple call
-    interceptor = TracingClientInterceptor()
-    config = client.config()
-    config["interceptors"] = [interceptor.intercept_client]
-    client = Client(**config)
-    handle = await client.start_workflow(
-        "kitchen_sink",
-        KSWorkflowParams(),
-        id=str(uuid.uuid4()),
-        task_queue=worker.task_queue,
-    )
-    await handle.result()
-
-    # Check trace
-    assert interceptor.traces[0][0] == "start_workflow"
-    assert interceptor.traces[0][1].workflow == "kitchen_sink"
-
-
 async def test_tls_config(tls_client: Optional[Client]):
     if not tls_client:
         pytest.skip("No TLS client")
     resp = await tls_client.workflow_service.describe_namespace(
-        temporalio.api.workflowservice.v1.DescribeNamespaceRequest(
-            namespace=tls_client.namespace
-        )
+        DescribeNamespaceRequest(namespace=tls_client.namespace)
     )
     assert resp.namespace_info.name == tls_client.namespace
+
+
+async def test_lazy_client(server: ExternalServer):
+    # Create another client that is lazy. This test just makes sure the
+    # functionality continues to work.
+    lazy_client = await Client.connect(
+        server.host_port, namespace=server.namespace, lazy=True
+    )
```

Comment:
Why does a lazy client require `await`ing? Is this a limitation of calling into the bridge?

Reply:
Yes, but it's an artificial limitation. I decided the API was simpler to just make `lazy` an option instead of making a whole new static method just for it. So even though there's technically no async going on during lazy, that it's on an async function means you have to await.

Reply:
I would prefer non async factory but it's just a nit.

Reply:
I definitely want to discourage use of this feature. But if enough people want this outside of an async context, I can make a way for that to happen without adding a static method here (basically using the constructor of this client directly and an additional non-static method for lazy creation on `temporalio.service.ServiceClient`).